src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py

```diff
@@ -0,0 +1,599 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge,
+# to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+ppo_tuner.py including:
+    class PPOTuner
+"""
+
+import os
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
+import copy
+import logging
+import numpy as np
+import json_tricks
+from gym import spaces
+
+import nni
+from nni.tuner import Tuner
+from nni.utils import OptimizeMode, extract_scalar_reward
+
+from .model import Model
+from .util import set_global_seeds
+from .policy import build_lstm_policy
+
+
+logger = logging.getLogger('ppo_tuner_AutoML')
+
+def constfn(val):
+    """wrap as function"""
+    def f(_):
+        return val
+    return f
+
+
+class ModelConfig:
+    """
+    Configurations of the PPO model
+    """
+    def __init__(self):
+        self.observation_space = None
+        self.action_space = None
+        self.num_envs = 0
+        self.nsteps = 0
+
+        self.ent_coef = 0.0
+        self.lr = 3e-4
+        self.vf_coef = 0.5
+        self.max_grad_norm = 0.5
+        self.gamma = 0.99
+        self.lam = 0.95
+        self.cliprange = 0.2
+        self.embedding_size = None  # the embedding is for each action
```

Comment:
I think we should decide whether the comment is placed on the right or on top?

Reply:
sure, let's schedule a short meeting for this.